### PR TITLE
[codestyle] Reduce Cyclomatic Complexity of method

### PR DESCRIPTION
**Pull request by ApCoder123**
Resolves issues:
http://sonar.xwiki.org/issues/search#issues=6b75d251-a45a-4391-b726-58c47888f330
http://sonar.xwiki.org/issues/search#issues=9a20df47-672f-4fca-9a38-767978c7d7e6

### CODE REVIEW

File: xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/XWikiSyntaxChainingRenderer.java [hunk 1]
```diff
@@ -223,20 +223,15 @@ public void endLink(ResourceReference reference, boolean freestanding, Map<Strin
         }
     }
 
-    @Override
+        @Override
```
Comment (tmortagne) [inline xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/XWikiSyntaxChainingRenderer.java]: Looks like a mistake.

File: xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/XWikiSyntaxChainingRenderer.java [hunk 2]
```diff
@@ -223,20 +223,15 @@ public void endLink(ResourceReference reference, boolean freestanding, Map<Strin
...
     public void beginFormat(Format format, Map<String, String> parameters)
     {
         // If the previous format had parameters and the parameters are different from the current ones then close them
         if (this.previousFormatParameters != null) {
-            if (parameters.isEmpty()) {
-                // print("(%%)");
-                // this.previousFormatParameters = null;
-            } else if (!this.previousFormatParameters.equals(parameters)) {
-                this.previousFormatParameters = null;
+            if (!this.previousFormatParameters.equals(parameters)) {
```
Comment (tmortagne) [inline xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/XWikiSyntaxChainingRenderer.java]: If previousFormatParameters is empty you will enter this if while it used to not be the case before.

File: xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/XWikiSyntaxChainingRenderer.java (hunk 1 above)
Comment (ApCoder123) [reply xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/XWikiSyntaxChainingRenderer.java]: Ok. Fixed

File: xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/XWikiSyntaxChainingRenderer.java (hunk 2 above)
Comment (ApCoder123) [reply xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/XWikiSyntaxChainingRenderer.java]: Not sure what you mean here. Is something like this correct?
```

Comment (tmortagne) [reply xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/XWikiSyntaxChainingRenderer.java]: I mean that after your change you won't have the same behavior if both previousFormatParameters and parameters are empty.

**Comment by ApCoder123** [comment]
@tmortagne I've edited it so that the behaviour stays the same and the complexity is reduced

File: xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/XWikiSyntaxChainingRenderer.java [hunk 3]
```diff
@@ -227,16 +227,11 @@ public void endLink(ResourceReference reference, boolean freestanding, Map<Strin
...
-                // this.previousFormatParameters = null;
-            } else if (!this.previousFormatParameters.equals(parameters)) {
-                this.previousFormatParameters = null;
+        if (this.previousFormatParameters != null && !parameters.isEmpty()) {
+            if (!this.previousFormatParameters.equals(parameters)) {
                 printParameters(parameters, false);
-            } else {
-                this.previousFormatParameters = null;
             }
+                this.previousFormatParameters = null;
```
Comment (tmortagne) [inline xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/XWikiSyntaxChainingRenderer.java]: The indentation is not right.

**Comment by ApCoder123** [comment]
Ok fixed the indentation

**Comment by tmortagne** [comment]
Thanks for your contribution !

**Comment by tmortagne** [comment]
Actually it makes several test fail so it does change something I did not noticed by reading the code. You can reproduce by building xwiki-rendering-integration-tests module. Reverting.

**Pull request merged**